source-test/File.ts

```diff
@@ -1,661 +1,1617 @@
+      const file = new Vinyl({ contents: null });
+      expect(file.isBuffer()).toEqual(false);
       done();
     });
   });
 
-  describe('isStream()', () => {
-    it('should return false when the contents are a Buffer', done => {
+  describe('isStream()', function() {
+
+    it('returns false when the contents are a Buffer', function(done) {
       const val = new Buffer('test');
-      const file = new File({ contents: val });
-      file.isStream().should.equal(false);
+      const file = new Vinyl({ contents: val });
+      expect(file.isStream()).toEqual(false);
       done();
     });
 
-    it('should return true when the contents are a Stream', done => {
-      const file = new File({ contents: fakeStream });
-      file.isStream().should.equal(true);
+    it('returns true when the contents are a Stream', function(done) {
+      const val = from([]);
+      const file = new Vinyl({ contents: val });
+      expect(file.isStream()).toEqual(true);
       done();
     });
 
-    it('should return false when the contents are a null', done => {
-      const file = new File({ contents: null });
-      file.isStream().should.equal(false);
+    it('returns false when the contents are null', function(done) {
+      const file = new Vinyl({ contents: null });
+      expect(file.isStream()).toEqual(false);
       done();
     });
   });
 
-  describe('isNull()', () => {
-    it('should return false when the contents are a Buffer', done => {
+  describe('isNull()', function() {
+
+    it('returns false when the contents are a Buffer', function(done) {
       const val = new Buffer('test');
-      const file = new File({ contents: val });
-      file.isNull().should.equal(false);
+      const file = new Vinyl({ contents: val });
+      expect(file.isNull()).toEqual(false);
       done();
     });
 
-    it('should return false when the contents are a Stream', done => {
-      const file = new File({ contents: fakeStream });
-      file.isNull().should.equal(false);
+    it('returns false when the contents are a Stream', function(done) {
+      const val = from([]);
+      const file = new Vinyl({ contents: val });
+      expect(file.isNull()).toEqual(false);
       done();
     });
 
-    it('should return true when the contents are a null', done => {
-      const file = new File({ contents: null });
-      file.isNull().should.equal(true);
+    it('returns true when the contents are null', function(done) {
+      const file = new Vinyl({ contents: null });
+      expect(file.isNull()).toEqual(true);
       done();
     });
   });
 
-  describe('clone()', () => {
-    it('should copy all attributes over with Buffer', done => {
-      const options = {
-        cwd: '/',
-        base: '/test/',
-        path: '/test/test.coffee',
-        contents: new Buffer('test')
-      };
-      const file = new File(options);
-      const file2 = file.clone();
+  describe('isDirectory()', function() {
+    const fakeStat: fs.Stats = <any> {
```

Comment:
Node's `Stats` constructor is not even documented. I could use it ([see the source](https://github.com/nodejs/node/blob/master/lib/fs.js#L132)) but which values should I pass it ?
Vinyl tests a real `Stats` object [only once](https://github.com/gulpjs/vinyl/blob/master/test/file.js#L601).

Since Vinyl is supposed to handle "virtual" files, I kinda understand that its tests do not completely match what is documented: it relies on knowledge of which internal methods are called. If you look trough the code, it sometimes defines only `isDirectory`, sometimes `isSymbolicLink`, sometimes both. I agree that it would be better if he only used real `Stats` object since it is documented as this, but do we really need to change it if we are never going to run the tests ?